src/Microsoft.PowerShell.Commands.Management/commands/management/TestConnectionCommand.cs

```diff
@@ -285,47 +290,91 @@ protected override void StopProcessing()
 
         private void ProcessConnectionByTCPPort(string targetNameOrAddress)
         {
+            if (Repeat.IsPresent)
+            {
+                Count = int.MaxValue;
+            }
+            else if (!MyInvocation.BoundParameters.ContainsKey("Count"))
+            {
+                Count = 1;
+            }
+
             if (!TryResolveNameOrAddress(targetNameOrAddress, out _, out IPAddress? targetAddress))
             {
                 return;
             }
 
-            TcpClient client = new TcpClient();
-
-            try
+            for (var i = 1; i <= Count; i++)
             {
-                Task connectionTask = client.ConnectAsync(targetAddress, TcpPort);
-                string targetString = targetAddress.ToString();
+                TcpPortStatus testResult = new TcpPortStatus(
+                    i,
+                    Source,
+                    targetNameOrAddress,
+                    targetAddress,
+                    TcpPort,
+                    0,
+                    false
+                );
 
-                for (var i = 1; i <= TimeoutSeconds; i++)
+                Stopwatch stopwatch = new Stopwatch();
+
+                using TcpClient client = new TcpClient();
+
+                try
                 {
-                    Task timeoutTask = Task.Delay(millisecondsDelay: 1000);
-                    Task.WhenAny(connectionTask, timeoutTask).Result.Wait();
+                    stopwatch.Start();
 
-                    if (timeoutTask.Status == TaskStatus.Faulted || timeoutTask.Status == TaskStatus.Canceled)
+                    if (client.ConnectAsync(targetAddress, TcpPort).Wait(TimeoutSeconds * 1000))
                     {
-                        // Waiting is interrupted by Ctrl-C.
-                        WriteObject(false);
-                        return;
+                        testResult.Connected = true;
+                        testResult.Latency = stopwatch.ElapsedMilliseconds;
                     }
+                    else
+                    {
+                        testResult.Status = SocketError.TimedOut;
```

Comment:
Again we need the connection status, not task status.
I think we get the status with exception.

Reply:
Timeouts are considered part of normal / expected output in the ICMP tests; is there reason to treat them as exceptions here?

Reply:
It comes from the API - if socket in TCPClient timeouts it throws.

Reply:
Aye, but the point of the test is already to determine if it's accessible or not. A timeout seems like an expected and forseeable outcome of such a test. Throwing is appropriate when that is expected to be the exception rather than the norm, but `Test-Connection` is not making an assumption that an endpoint is accessible; it's checking.

Given that, I think it's more appropriate to simply report a status rather than throw an error.

Reply:
It seems we discuss this in #11517 too. Let's wait #11517.